src/components/review/ShopSearch.tsx

```diff
@@ -0,0 +1,127 @@
+import { shopApi } from 'features/shops/shopApi';
+import Image from 'next/image';
+import { useRouter } from 'next/router';
+import searchDelIC from 'public/assets/ic_searchDel.svg';
+import { ChangeEvent, FormEvent, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { theme } from 'styles/theme';
+import { Shop } from 'types/shop';
+
+import ShopSearchList from './ShopSearchList';
+
+function ShopSearch() {
+  const [inputValue, setInputValue] = useState('');
+  const [searchValue, setSearchValue] = useState<Shop[] | undefined>();
+  const [trigger] = shopApi.useLazyGetShopSearchResultQuery();
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedShop, setSelectedShop] = useState('');
+  const toggle = () => setIsOpen((prevIsOpen) => !prevIsOpen);
+  const router = useRouter();
+  const { shopName } = router.query;
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    if (!(e.target instanceof HTMLInputElement)) return;
+    setInputValue(e.target.value);
+  };
+
+  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
+    if (!(e.target instanceof HTMLFormElement)) return;
+    e.preventDefault();
+    const result = await trigger(inputValue);
+    setSearchValue(result.data);
+    setIsOpen(true);
+  };
```

Comment:
```suggestion
  const requestSearch = async () => {
    const result = await trigger(inputValue);
    setSearchValue(result.data);
    setIsOpen(true);
  };

  const handleSubmit = async (e: FormEvent<HTMLFormElement>) => {
    if (!(e.target instanceof HTMLFormElement)) return;
    e.preventDefault();
    await requestSearch();
  };
```

이런 식으로 기존의 submit 핸들러를 분리하고 `requestSearch`를 검색 아이콘의 온클릭 핸들러로 등록해주면 될 것 같아요!